react/week3/src/TodosFetcher.js

```diff
@@ -0,0 +1,27 @@
+import React from "react";
+
+function fetcher() {
+  const todosURL =
+    "https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw";
+  return fetch(todosURL).then((response) => response.json());
+}
+
+class TodosFetcher extends React.Component {
+state = {
+ todo: " ",
+ dueDate: " "   
+}
+
+componentDidMount() {
+    this.fetcher
+}
+
+
+    render() {
+        return (
+            <div>
+                
+            </div>
+        )
+    }
+}
```

Comment:
This seems incomplete. If async code in React is causing you trouble, give this video a run through and see if it helps improve your understanding: https://www.youtube.com/watch?v=T3Px88x_PsA

Reply:
Yes! This is incomplete because I got stuck in some parts of the checklist. But your video comes like water in the dessert to help. Thanks!

Reply:
Hi Braedon, 
Code is much more clear in this way. The only issue I would like to review is that i can't render the data to fit it in the setState. 
PR d8ed2b187fc41a53c1604b9a5c3d69079e2e1d5f
```  
async componentDidMount() {
    const url =      "https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw";
    const response = await fetch(url);
    const data = await response.json();
    this.setState({ todos: data, complete: false, isEdit: false });
  }